### ISSUE TITLE
collect on independent variable in ODE systems

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#16894]
It might be worth looking at including expansion and collection on the independent variable in system of ODEs. Consider `sol8` of "test_linear_2eq_order2"
```python
>>> count_ops(sol8)
183
>>> count_ops([Eq(i.lhs, collect(i.rhs.expand(), t)) for i in sol8])
51
```

**Comment by oscarbenjamin** [comment]
The solutions are created here:
https://github.com/sympy/sympy/blob/fb7bdbc7842623f3ed5dbead9e751b9468836dff/sympy/solvers/ode.py#L7647-L7652

A bit of expansion and collection on sol1 and sol2 would probably help some of the time.

The separated equations being solved recursively by dsolve are both Airy's equation. We wouldn't have the series solution if Airy's equation was properly solved for as in #16581 which gives the solution as:
```python
C1, C2 = symbols('C1, C2') 
sol8 = [Eq(x(t), -sqrt(133)*(-4*C1*airyai(t*(-1 + sqrt(133))**(S(1)/3)) + 4*C1*airyai(-t*(1 + \
sqrt(133))**(S(1)/3)) - 4*C2*airybi(t*(-1 + sqrt(133))**(S(1)/3)) + 4*C2*airybi(-t*(1 + sqrt(133))**(S(1)/3)) +\
(-sqrt(133) - 1)*(C1*airyai(t*(-1 + sqrt(133))**(S(1)/3)) + C2*airybi(t*(-1 + sqrt(133))**(S(1)/3))) - (-1 +\
sqrt(133))*(C1*airyai(-t*(1 + sqrt(133))**(S(1)/3)) + C2*airybi(-t*(1 + sqrt(133))**(S(1)/3))))/3192), \
Eq(y(t), -sqrt(133)*(-C1*airyai(t*(-1 + sqrt(133))**(S(1)/3)) + C1*airyai(-t*(1 + sqrt(133))**(S(1)/3)) -\
C2*airybi(t*(-1 + sqrt(133))**(S(1)/3)) + C2*airybi(-t*(1 + sqrt(133))**(S(1)/3)))/266)]
```
With that we have:
```julia
In [13]: count_ops(sol8)                                                                                                          
Out[13]: 139

In [14]: count_ops([Eq(i.lhs, collect(i.rhs.expand(), t)) for i in sol8])                                                         
Out[14]: 156
```

Actually this type6 solver should be fixed so that it doesn't mix together the integration constants from the two different calls to dsolve. The general solution here should have 4 distinct integration constants (integration constants should be like dummy symbols).

**Comment by smichr** [comment]
Thanks for the feedback, @oscarbenjamin
